src/codecs/ready_frames.rs

```diff
@@ -0,0 +1,113 @@
+use std::pin::Pin;
+
+use futures::{
+    task::{Context, Poll},
+    {Stream, StreamExt},
+};
+
+const DEFAULT_CAPACITY: usize = 1024;
+
+/// A stream combinator aimed at improving the performance of decoder streams under load.
+///
+/// This is similar in spirit to `StreamExt::ready_chunks`, but built specifically for the
+/// particular result tuple returned by decoding streams. The more general `FoldReady` is left as
+/// an exercise to the reader.
+pub struct ReadyFrames<T, U, E> {
```

Comment:
I realize we are probably testing this at a higher level, but it strikes me as the sort of type it'd be nice to have some unit tests around too.